src/account.ts

```diff
@@ -95,10 +99,19 @@ export class Erc20Coin extends AccountCoinToken {
  * the token code and the issuer account in the form: (t)xlm:<token>-<issuer>
  */
 export class StellarCoin extends AccountCoinToken {
-  constructor(options: AccountConstructorOptions) {
+  public domain: string;
+
+  constructor(options: StellarCoinConstructorOptions) {
     super({
       ...options,
     });
+
+    const domainPattern = /^(?:[a-z0-9](?:[a-z0-9-]{0,61}[a-z0-9])?\.)+[a-z0-9][a-z0-9-]{0,61}[a-z0-9]$/;
+    if (options.domain !== '' && !options.domain.match(domainPattern)) {
+      throw new InvalidDomain(options.name, options.domain);
+    }
+
+    this.domain = (options.domain as unknown) as string;
```

Comment:
I don't think we need the casting craziness from Erc20Coin, since `this.domain` and `options.domain` are both strings.

Reply:
good, I was confused by it. Fixed